flink-clients/src/main/java/org/apache/flink/client/RemoteExecutor.java

```diff
@@ -226,32 +226,6 @@ public String getOptimizerPlanAsJSON(Plan plan) throws Exception {
 
 	@Override
 	public void endSession(JobID jobID) throws Exception {
-		if (jobID == null) {
```

Comment:
(1/2) Remove `endSession(JobID)` use point. Please check that this doesn't change current behaviors.